pkg/deployer/utils/utils.go

```diff
@@ -0,0 +1,60 @@
+// SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package utils
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	lsv1alpha1 "github.com/gardener/landscaper/apis/core/v1alpha1"
+	lsv1alpha1helper "github.com/gardener/landscaper/apis/core/v1alpha1/helper"
+)
+
+// HandleAnnotationsAndGeneration is meant to be called at the beginning of a deployer's reconcile loop.
+// If a reconcile is needed due to the reconcile annotation or a change in the generation, it will set the phase to Init and remove the reconcile annotation.
+// It will also remove the timeout annotation if it is set.
+// Returns:
+//   - the modified deployitem
+//   - an error, if updating the deployitem failed, nil otherwise
+func HandleAnnotationsAndGeneration(ctx context.Context, c client.Client, log logr.Logger, di *lsv1alpha1.DeployItem) error {
+	changedMeta := false
+	changedStatus := false
```

Comment:
these 2 variables seems unnecessary.
Why not just update the status after line 37 and the item after line 42?

Reply:
Hm, we want to do as few updates as possible, because each one potentially causes conflicts with the deployers. The problem is that within the first `if` block, the metadata might or might not change, so we don't know for sure whether we need to update the object. If we update it directly after line 32, we might have to do it again if we go into the second `if` block, so we would have to update the object twice, which we are trying to avoid.

But I guess you are right about the status, this is only and always updated in the first `if`, so we could get rid of the `changedStatus` variable and do the update and the end of the first if.

WDYT?